PhysicsTools/Heppy/python/physicsutils/BTagWeightCalculator.py

```diff
@@ -33,34 +33,51 @@ def getHistosFromFile(self, fn):
         ROOT.gROOT.cd()
         for k in tf.GetListOfKeys():
             kn = k.GetName()
-            if not kn.startswith("csv_ratio"):
+            if not (kn.startswith("csv_ratio") or kn.startswith("c_csv_ratio") ):
                 continue
             spl = kn.split("_")
+            is_c = 1 if kn.startswith("c_csv_ratio") else 0
 
-            if spl[2] == "all":
+            if spl[2+is_c] == "all":
```

Comment:
please state in comments what the key should look like, so that people can understand the logic of your operations on the spl fields. Also, please avoid too short variable names. e.g. is_c means little by itself, is_charm would be better.